packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -639,17 +639,20 @@ class StripeHelper {
    * @param {Subscription[id]} subscriptionId
    */
   async cancelSubscriptionForCustomer(uid, email, subscriptionId) {
-    const hasSubscription = await this.subscriptionForCustomer(
+    const subscription = await this.subscriptionForCustomer(
       uid,
       email,
       subscriptionId
     );
-    if (!hasSubscription) {
+    if (!subscription) {
       throw error.unknownSubscription();
     }
 
     await this.stripe.subscriptions.update(subscriptionId, {
       cancel_at_period_end: true,
+      metadata: {
```

Comment:
So, this will over-write any exiting metadata. To preserve any additional properties, the entire metadata object needs to be passed in, with the appropriate field updated.

Reply:
Hmm, I'd had this as a merge with original metadata earlier. But then this bit of [the Stripe docs](https://stripe.com/docs/api/subscriptions/update#update_subscription-metadata) made me think this was merged in by the API:

> Individual keys can be unset by posting an empty value to them. All keys can be unset by posting an empty value to metadata.

I can tweak this to do the merge with existing metadata here though